api/src/controllers/login.js

```diff
@@ -137,7 +137,12 @@ const setCookies = (req, res, sessionRes) => {
     .then(userCtx => {
       setSessionCookie(res, sessionCookie);
       setUserCtxCookie(res, userCtx);
-      res.json({ success: true });
+      // https://github.com/medic/medic-webapp/issues/5035
+      //  For Test DB, temporarily disable `canCongifure` property to avoid redirecting to admin console
+      // One `e2e` is problematic 
+      const designDoc  = auth.hasAllPermissions(userCtx, 'can_configure') &&
+        environment.db !== 'medic-test' ? 'medic-admin' : 'medic';
+      res.status(307).send(path.join('/', environment.db, '_design', designDoc, '_rewrite'));
```

Comment:
You're sending the path in the body of the response but to be "correct" it should be in the `Location` header. Also, I made a mistake - I think a 302 is better than a 307 here and it's more common as well so we should use that. So, I think this code should be...

```js
res.redirect(path.join(...));
```

Then the login/script.js should be something like this:

```js
window.location = xmlhttp.getResponseHeader('Location');
```

Reply:
The only challenge here is that we are using `XMLHTTPrequest` to negotiate communication between the server and the client. In the event that I do

```js
res.redirect(path.join(...));
```

XMLHTTPrequest will default to response code `200` and throw the html content for `res.redirect` as an error `xmlhttp.response`  i.e. basically not rendering the html on the browser

Reply:
Wow, ok, it looks like `XMLHttpRequest` automatically follows the 302... I think that's an error in implementation but there's nothing we can do about that.

Your implementation is fine - stick with it. Good job!